prettyprinter/src/Data/Text/Prettyprint/Doc/Internal.hs

```diff
@@ -553,7 +553,7 @@ changesUponFlattening :: Doc ann -> Maybe (Doc ann)
 changesUponFlattening = \doc -> case doc of
     FlatAlt _ y     -> Just (flatten y)
     Line            -> Just Fail
-    Union x _       -> changesUponFlattening x <|> Just x
+    Union x _       -> Just x
```

Comment:
Another idea:

```suggestion
    Union x _       -> Nothing
```

Reply:
Ah, no. This causes the fusion tests to fail with the classic

> »SFail« must not appear in a rendered »SimpleDocStream«.

I'm not quite sure why though. Proper shrinking and diagnostic output would help… :/